crates/errors/src/fmt.rs

```diff
@@ -27,50 +27,101 @@
 //! # }
 //! ```
 
-use codederror::{Code, CodedError};
+use codederror::Code;
 use std::fmt;
 
 /// Check module documentation for more details.
 #[macro_export]
 macro_rules! info_it {
     ($err:expr) => {
-        tracing::info!(error = tracing::field::display(codederror::CodedError::decorate(&$err)), restate.error.code = ?$crate::fmt::RestateCode::from(&$err));
+        {
+            #[allow(unused_imports)]
+            use ::codederror::CodedError;
+
+            let err = &$err;
+            let decorated = err.decorate();
+            let code = $crate::fmt::RestateCode::from_code(err.code());
+
+            tracing::info!(error = tracing::field::display(decorated), restate.error.code = ?code);
+        }
```

Comment:
Doesn't really change any logic, but because of how i call `decorate()` and `code()` as receiver methods, now it works both with `&err` and `&&err`, which is what i was trying to achieve with this change.